### PR TITLE
[Enhancement] assign tablets to driver sequence only if num_buckets>1

### PR DESCRIPTION
**Pull request by ZiheLiu**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

## Problem Summary(Required) ：

As for a local bucket shuffle join, we assign tablets to each driver sequence, and set `DOP` to the number of buckets a BE is assigned, in order to avoid local shuffle.

When enabling tablet internal parallel, we prefer setting `numBuckets` to `numBEs`, that is , each BE has **one** bucket.
Therefore, `DOP` of local bucket shuffle join in this scenario is only **one**.

In this case, we should insert local shuffle to improve the degree of parallelism of JoinOperator. Therefore, apply the strategy about assign tablets to each driver sequence, only if `num_buckets_of_one_BE` > 1. 

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by ZiheLiu** [comment]
run all

**Comment by ZiheLiu** [comment]
run starrocks_admit_test

**Comment by ZiheLiu** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/qe/Coordinator.java [hunk 1]
```diff
@@ -2017,16 +2022,24 @@ public void computeColocatedJoinInstanceParam(Map<Integer, TNetworkAddress> buck
                 }
                 List<List<Map.Entry<Integer, Map<Integer, List<TScanRangeParams>>>>> scanRangesPerDriverSeq =
                         ListUtil.splitBySize(scanRangePerInstance, expectedDop);
-                instanceParam.pipelineDop = scanRangesPerDriverSeq.size();
+
+                boolean assignPerDriverSeq = enablePipeline && scanRangesPerDriverSeq.size() > 1;
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/qe/Coordinator.java:2026]: Add your PR comment to the code comment to explain why need to check ` scanRangesPerDriverSeq.size() > 1`

**Comment by ZiheLiu** [comment]
@Mergifyio backport branch-2.4

**Comment by ZiheLiu** [comment]
run starrocks_admit_test

**Review by kangkaisen — DISMISSED** [review]

**Comment by ZiheLiu** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 11 / 15 (73.33%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|11|15|73.33%|[1448, 1449, 1521, 1522]|

**Comment by ZiheLiu** [comment]
use starrocks_admit_test

**Comment by ZiheLiu** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 11 / 15 (73.33%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|11|15|73.33%|[1448, 1449, 1521, 1522]|

**Comment by ZiheLiu** [comment]
run starrocks_admit_test

**Comment by ZiheLiu** [comment]
run starrocks_admit_test

**Comment by mofeiatwork** [comment]
run all

**Comment by ZiheLiu** [comment]
run all

**Comment by ZiheLiu** [comment]
run all

**Comment by ZiheLiu** [comment]
run starrocks_admit_test

**Comment by ZiheLiu** [comment]
run all

**Comment by ZiheLiu** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 15 / 19 (78.95%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|15|19|78.95%|[1448, 1449, 1521, 1522]|

**Comment by ZiheLiu** [comment]
run starrocks_admit_test

**Comment by ZiheLiu** [comment]
run starrocks_admit_test

**Comment by ZiheLiu** [comment]
run starrocks_fe_unittest

**Comment by ZiheLiu** [comment]
run starrocks_admit_test

**Comment by ZiheLiu** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 15 / 19 (78.95%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|15|19|78.95%|[1448, 1449, 1521, 1522]|

**Review by kangkaisen — APPROVED** [review]

**Review by satanson — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by ZiheLiu** [comment]
run starrocks_admit_test

**Comment by ZiheLiu** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 15 / 19 (78.95%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|15|19|78.95%|[1448, 1449, 1521, 1522]|

**Pull request merged**